cli/azd/pkg/azapi/standard_deployments.go

```diff
@@ -691,3 +694,110 @@ func convertFromStandardProvisioningState(state armresources.ProvisioningState)
 
 	return DeploymentProvisioningState("")
 }
+
+func (ds *StandardDeployments) ValidatePreflightToSubscription(
+	ctx context.Context,
+	subscriptionId string,
+	location string,
+	deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+	tags map[string]*string,
+	options map[string]any,
+) error {
+	deploymentClient, err := ds.createDeploymentsClient(ctx, subscriptionId)
+	if err != nil {
+		return fmt.Errorf("creating deployments client: %w", err)
+	}
+
+	var rawResponse *http.Response
+	ctx = policy.WithCaptureResponse(ctx, &rawResponse)
+
+	validateResult, err := deploymentClient.BeginValidateAtSubscriptionScope(
+		ctx, deploymentName,
+		armresources.Deployment{
+			Properties: &armresources.DeploymentProperties{
+				Template:   armTemplate,
+				Parameters: parameters,
+				Mode:       to.Ptr(armresources.DeploymentModeIncremental),
+			},
+			Location: to.Ptr(location),
+			Tags:     tags,
+		}, nil)
+	if err != nil {
+		return validatePreflightError(rawResponse, err, "subscription")
+	}
+	_, err = validateResult.PollUntilDone(ctx, nil)
+	if err != nil {
+		return validatePreflightError(rawResponse, err, "subscription")
+	}
+
+	return nil
+}
+
+func validatePreflightError(
+	rawResponse *http.Response,
+	err error,
+	typeMessage string,
+) error {
+	var respErr *azcore.ResponseError
+	if errors.As(err, &respErr) {
+		err = createDeploymentError(err)
+	} else if err != nil {
+		// Error returned from azure sdk go bug: we receive a 400 Bad Request from the API,
+		// but the client-handling in azure sdk fails internally with a different error
+		// This special-cased handling and rawResponse capture can be removed once
+		// https://github.com/Azure/azure-sdk-for-go/issues/23350 is fixed
+		if rawResponse != nil && rawResponse.StatusCode == 400 {
+			defer rawResponse.Body.Close()
+			body, errOnRawResponse := io.ReadAll(rawResponse.Body)
+			if errOnRawResponse != nil {
+				return fmt.Errorf("failed to read response error body from preflight validation api to %s: %w",
+					typeMessage, errOnRawResponse)
+			}
+
+			err = NewAzureDeploymentError(string(body))
+		}
+	}
+	return fmt.Errorf(
+		"validating preflight to %s:\n\nPreflight Error Details:\n%w",
```

Comment:
Would something like this look any better?

"validating deployment:\n\nValidation errors:\n"